x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/alerts.ts

```diff
@@ -427,6 +452,189 @@ export default function alertTests({ getService }: FtrProviderContext) {
               throw new Error(`Scenario untested: ${JSON.stringify(scenario)}`);
           }
         });
+
+        it('should throttle alerts when appropriate', async () => {
+          const reference = `create-test-5:${user.username}`;
+          const createdAction = await createIndexRecordAction(space.id);
+          const response = await supertestWithoutAuth
+            .post(`${getUrlPrefix(space.id)}/api/alert`)
+            .set('kbn-xsrf', 'foo')
+            .auth(user.username, user.password)
+            .send(
+              getTestAlertData({
+                interval: '1s',
+                throttle: '1m',
+                alertTypeId: 'test.always-firing',
+                alertTypeParams: {
+                  index: esTestIndexName,
+                  reference,
+                },
+                actions: [
+                  {
+                    group: 'default',
+                    id: createdAction.id,
+                    params: {
+                      index: esTestIndexName,
+                      reference,
+                      message: '',
+                    },
+                  },
+                ],
+              })
+            );
+
+          switch (scenario.id) {
+            case 'no_kibana_privileges at space1':
+            case 'space_1_all at space2':
+            case 'global_read at space1':
+              expect(response.statusCode).to.eql(404);
+              expect(response.body).to.eql({
+                statusCode: 404,
+                error: 'Not Found',
+                message: 'Not Found',
+              });
+              break;
+            case 'space_1_all at space1':
+            case 'superuser at space1':
+              // Wait until alerts fired 3 times to ensure actions had a chance to fire twice
+              await waitForTestIndexDocs('alert:test.always-firing', reference, 3);
```

Comment:
fyi: I've noticed in practice that alert executors - or maybe action executors for alerts - get run at ~2s + specified interval, so this will be waiting for 9 seconds.  Time to fix that issue, perhaps, because there's no other way to test this! :-)

Reply:
I've looked into this at some point. One fix is to shrink the delay when testing but I learned that when testing against Cloud, they only use the default config. It was recommended to stay as close as we can to how Cloud runs their tests. There's been a long time idea of changing task manager to make it easier for functional test. Maybe having extra APIs that can trigger tasks right away or something. It was just missing a backlog card that I just created (https://github.com/elastic/kibana/projects/26#card-26403818).